### PR TITLE
Add text tracks for closed captions listed in playlist

### PR DESCRIPTION
**Pull request by squarebracket**
## Description
Currently, text tracks for captions are created by
videojs-contrib-media-sources. With the open PRs for multiple caption
track support in [mux.js](https://github.com/videojs/mux.js/pull/150)
and [contrib-media-sources](https://github.com/videojs/videojs-contrib-media-sources/pull/140)
text tracks are simply created using their caption track (`CCn`). This
PR creates tracks at playlist read time. Afterwards, contrib-media-sources will look to see if a text track that has the ID of `CCn` exists, and if so, it will use that track.

## Specific Changes proposed
Right after subtitle text tracks are created, closed caption text tracks are created.

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [x] If necessary, more likely in a feature request than a bug fix
  - [x] Unit Tests updated or fixed
  - [ ] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](http://jsbin.com/liwecukasi/edit?html,output))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by mjneil** [comment]
Blocked on
https://github.com/videojs/videojs-contrib-media-sources/pull/140
and
https://github.com/videojs/mux.js/pull/150

File: src/master-playlist-controller.js [hunk 1]
```diff
@@ -816,6 +818,43 @@ export class MasterPlaylistController extends videojs.EventTarget {
...
+          videojs.mergeOptions({ id: label }, properties));
+
+        if (typeof this.closedCaptionGroups_.tracks[label] === 'undefined') {
+          let track = this.tech_.addRemoteTextTrack({
+            id: properties.instreamId,
+            kind: 'captions',
+            enabled: false,
+            language: properties.language,
+            label
+          }, true).track;
```
Comment (mjneil) [inline src/master-playlist-controller.js]: If we change this parameter from `true` to `false`, video.js will take care of removing the track on source changes

**Comment by squarebracket** [comment]
just found a problem -- need to ignore CEA708 tracks since we don't support them.

**Comment by mjneil** [comment]
Oh those can be specified in manifest as well. Is there a way to tell them apart in manifest?

**Comment by squarebracket** [comment]
Yeah, by inspecting the value of `INSTREAM-ID`. For 608 the values are CC1-CC4, for 708 the values are SERVICE1-SERVICE-64. We can just do a regex match.

File: src/master-playlist-controller.js (hunk 1 above)
Comment (mjneil) [inline src/master-playlist-controller.js]: bump :)

Comment (squarebracket) [inline src/master-playlist-controller.js]: derp derp, thanks.

**Comment by squarebracket** [comment]
I'm doing some manual testing and I'm getting a 708 track added, looking into it...

**Comment by squarebracket** [comment]
Nevermind.. Was mistakenly using wrong code :roll_eyes:

**Pull request merged**